From aa7fc1c9d627957e5c7df8a90e8613d8d4f78069 Mon Sep 17 00:00:00 2001
From: Ryan Zezeski <ryan@zinascii.com>
Date: Thu, 13 Mar 2025 19:46:46 -0400
Subject: [PATCH] fix watch init

I'm not sure how this was ever supposed to compile on platforms that
don't implement the Watch API.
diff -wpruN --no-dereference '--exclude=*.orig' a~/lib/std/Build/Watch.zig a/lib/std/Build/Watch.zig
--- a~/lib/std/Build/Watch.zig	1970-01-01 00:00:00
+++ a/lib/std/Build/Watch.zig	1970-01-01 00:00:00
@@ -811,7 +811,7 @@ const Os = switch (builtin.os.tag) {
 };
 
 pub fn init() !Watch {
-    return Os.init();
+    return if (Os != void) Os.init() else error.Unsupported;
 }
 
 pub const Match = struct {
